### PR TITLE
feat(forms) Add validators min and max with their tests.

### PR DESCRIPTION
**Pull request by AnObstacle**
**What kind of change does this PR introduce?** (check one with "x")
```
[x] Feature
```

**What is the new behavior?**
Add min max validation for numeric input (like min max in html input)

**Does this PR introduce a breaking change?** (check one with "x")
```
[x] No
```

### CODE REVIEW

File: modules/@angular/forms/test/validators_spec.ts [hunk 1]
```diff
@@ -119,6 +119,66 @@ export function main() {
...
 
+    describe('min', () => {
+      it('should not error on an empty string',
+        () => { expect(Validators.min(2)(new FormControl(''))).toBeNull(); });
+
+      it('should not error on null',
+        () => { expect(Validators.min(2)(new FormControl(null))).toBeNull(); });
+
+      it('should not error on undefined',
+        () => { expect(Validators.min(2)(new FormControl(null))).toBeNull(); });
```
Comment (laco0416) [inline modules/@angular/forms/test/validators_spec.ts]: mistake of `new FormControl(undefined)`?

Comment (Mathou54) [inline modules/@angular/forms/test/validators_spec.ts]: Maybe, I confess I just copy and past the tests for minlength and maxlength. Then i change them to use the min and max function I have created.
Do I also change the null to undefined in the others tests?

Comment (laco0416) [inline modules/@angular/forms/test/validators_spec.ts]: I think you don't have to touch other tests.

Comment (Mathou54) [inline modules/@angular/forms/test/validators_spec.ts]: Ok, then it is done.

**Comment by mhevery** [comment]
@kara do we want to keep this? If so we would need to resurrect this PR, get it to pass and rebase.

File: modules/@angular/forms/test/validators_spec.ts [hunk 2]
```diff
@@ -125,6 +125,69 @@ export function main() {
...
+        expect(Validators.min(2)(new FormControl(1))).toEqual({
+          'min': {'min': 2, 'actual': 1}
+        });
+      });
+
+      it('should not error on big values', () => {
+        expect(Validators.min(2)(new FormControl(3))).toBeNull();
+      });
+
+      it('should not error on equals values', () => {
```
Comment (kara) [inline modules/@angular/forms/test/validators_spec.ts]: Typo: `equals` -> `equal` (same on line 185)

File: modules/@angular/forms/src/validators.ts [hunk 3]
```diff
@@ -100,6 +100,34 @@ export class Validators {
   }
 
   /**
+   * Validator that requires controls to have a value of a minimum number.
+   */
+  static min(min: number): ValidatorFn {
```
Comment (kara) [inline modules/@angular/forms/src/validators.ts]: This currently only works with reactive forms. Can we add the corresponding directives that provide these validators?

**Review by kara — CHANGES_REQUESTED** [review]
Thanks for the PR!  Can you add the validator directives and some template-driven integration tests for them? Then should be good.

**Comment by DzmitryShylovich** [comment]
@Mathou54 can you please update the commit message so it will automatically close https://github.com/angular/angular/issues/15053 ? thanks

**Comment by kara** [comment]
Looks like this PR has become stale :(.  I'm going to close for now since there is another PR for this feature that is more active.

**Pull request closed without merging**